### PR TITLE
Fixes #22852 - Python3 support for Facts plugin

### PR DESCRIPTION
**Pull request by jturel**
Like our yum plugins, we need to support RHEL8 here - so had to make it work on python 3.

I also fixed an incorrect exception, added some test coverage, and made the Makefile test targets a little more manageable.

### CODE REVIEW

File: src/rhsm-plugins/fqdn.py [hunk 1]
```diff
@@ -8,6 +8,6 @@ class FactsPlugin(SubManPlugin):
     name = 'fqdn fact'
 
     def post_facts_collection_hook(self, conduit):
-        if not conduit.facts.has_key('network.fqdn'):
```
Comment (jturel) [inline src/rhsm-plugins/fqdn.py]: has_key went missing from PY3

File: test-requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 mock
+unittest2
```
Comment (jturel) [inline test-requirements.txt]: unittest2 allows us to use newer unittest things in older python especially < 2.7 such as skipping test decorators

**Comment by jturel** [comment]
This can be tested with subman built for python3:

```
git clone https://github.com/candlepin/subscription-manager
dnf install -y tito npm GConf2-devel dbus-glib-devel gtk3-devel intltool libnotify-devel openssl-devel python3-devel redhat-lsb scrollkeeper python2-devel gcc python3-kitchen python3-tracer
npm install -g yarn
tito build --rpm --test
yum install /tmp/tito/x86_64/*.rpm -y
```
```
ln -s ~/katello-host-tools/src/rhsm-plugins/fqdn.py /usr/share/rhsm-plugins/fqdn.py
```

create /etc/rhsm/pluginconf.d/fqdn.FactsPlugin.conf with content
```
[main]
enabled=1
```

registering the system and checking its facts should show the proper fqdn of the system

**Comment by jlsherrill** [comment]
Thanks @jturel !

**Pull request merged**